spec/rubocop/cop/performance/collection_literal_in_loop_spec.rb

```diff
@@ -269,4 +269,74 @@
       RUBY
     end
   end
+
+  context 'when Ruby >= 3.4', :ruby34 do
+    # TODO: Remove when the minimum supported RuboCop is 1.62.0
+    let(:ruby_version) { 3.4 }
+
+    it 'registers an offense for `include?` on a Hash literal' do
+      expect_offense(<<~RUBY)
+        each do
+          { foo: :bar }.include?(:foo)
+          ^^^^^^^^^^^^^ Avoid immutable Hash literals in loops. It is better to extract it into a local variable or a constant.
+        end
+      RUBY
+    end
+
+    it 'registers an offense for other array methods' do
+      expect_offense(<<~RUBY)
+        each do
+          [1, 2, 3].index(foo)
+          ^^^^^^^^^ Avoid immutable Array literals in loops. It is better to extract it into a local variable or a constant.
+        end
+      RUBY
+    end
+
+    context 'when using an Array literal and calling `include?`' do
+      [
+        '"string"',
+        'self',
+        'local_variable',
+        'method_call',
+        '@instance_variable'
+      ].each do |argument|
+        it "registers no offense when the argument is #{argument}" do
+          expect_no_offenses(<<~RUBY)
+            #{'local_variable = 123' if argument == 'local_variable'}
```

Comment:
The presence or absence of this string interpolation does not change the testing result, but is this code meaningful in relation to the test's behavior?

Reply:
Removing it makes it send instead of lvar, but both are handled so the test doesn't visibly change